core/src/main/java/io/aiven/kafka/tieredstorage/metrics/Metrics.java

```diff
@@ -32,107 +33,259 @@
+            .with(metricsRegistry.segmentDeleteBytesTotalByTopicPartition, new CumulativeSum())
+            .get()
+            .record(bytes);
     }
 
-    public void recordSegmentCopyTime(final long startMs, final long endMs) {
-        segmentCopyTime.record(endMs - startMs);
+    private void recordSegmentDeleteRequests(final TopicPartition topicPartition) {
+        new SensorProvider(metrics, sensorName(SEGMENT_DELETE))
+            .with(metricsRegistry.segmentDeleteRequestsRate, new Rate())
+            .with(metricsRegistry.segmentDeleteRequestsTotal, new CumulativeCount())
+            .get()
+            .record();
+        new SensorProvider(metrics, sensorNameByTopic(topicPartition, SEGMENT_DELETE),
+            () -> MetricsRegistry.topicTags(topicPartition))
+            .with(metricsRegistry.segmentDeleteRequestsRateByTopic, new Rate())
+            .with(metricsRegistry.segmentDeleteRequestsTotalByTopic, new CumulativeCount())
+            .get()
+            .record();
+        new SensorProvider(metrics, sensorNameByTopicPartition(topicPartition, SEGMENT_DELETE),
+            () -> topicPartitionTags(topicPartition), Sensor.RecordingLevel.DEBUG)
+            .with(metricsRegistry.segmentDeleteRequestsRateByTopicPartition, new Rate())
+            .with(metricsRegistry.segmentDeleteRequestsTotalByTopicPartition, new CumulativeCount())
+            .get()
+            .record();
     }
 
-    public void recordSegmentCopyError() {
-        segmentCopyErrors.record();
+    public void recordSegmentDeleteTime(final TopicPartition topicPartition, final long startMs, final long endMs) {
+        final var time = endMs - startMs;
+        new SensorProvider(metrics, sensorName(SEGMENT_DELETE_TIME))
+            .with(metricsRegistry.segmentDeleteTimeAvg, new Avg())
+            .with(metricsRegistry.segmentDeleteTimeMax, new Max())
+            .get()
+            .record(time);
+        new SensorProvider(metrics, sensorNameByTopic(topicPartition, SEGMENT_DELETE_TIME),
+            () -> topicTags(topicPartition))
+            .with(metricsRegistry.segmentDeleteTimeAvgByTopic, new Avg())
+            .with(metricsRegistry.segmentDeleteTimeMaxByTopic, new Max())
+            .get()
+            .record(time);
+        new SensorProvider(metrics, sensorNameByTopicPartition(topicPartition, SEGMENT_DELETE_TIME),
+            () -> topicPartitionTags(topicPartition), Sensor.RecordingLevel.DEBUG)
+            .with(metricsRegistry.segmentDeleteTimeAvgByTopicPartition, new Avg())
+            .with(metricsRegistry.segmentDeleteTimeMaxByTopicPartition, new Max())
+            .get()
+            .record(time);
     }
 
-    public void recordSegmentDelete(final int bytes) {
-        segmentDeleteRequests.record();
-        segmentDeleteBytes.record(bytes);
+    public void recordSegmentDeleteError(final TopicPartition topicPartition) {
+        new SensorProvider(metrics, sensorName(SEGMENT_DELETE_ERRORS))
```

Comment:
Should probably be `SEGMENT_DELETE_ERRORS_RATE`? And below too

Reply:
If so, probably `SEGMENT_DELETE_ERRORS` and other "base" constants should be made private to prevent this misuse

Reply:
And also since the tests missed this too, maybe something is wrong with them

Reply:
It's correct actually. `SEGMENT_DELETE_ERRORS` is the sensor name, and `SEGMENT_DELETE_ERRORS_RATE` the metric name. `SEGMENT_DELETE_ERRORS_RATE` may be private. I may also add `METRIC_` and `SENSOR_` to make these names less confusing

Reply:
No, I'm wrong :)